StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift

```diff
@@ -480,7 +484,86 @@ extension STPAPIClient: FinancialConnectionsAPIClient {
 
     // MARK: - Networking
 
-    func saveAccountsToLink(
+    func saveAccountsToNetworkAndLink(
+        shouldPollAccounts: Bool,
+        selectedAccounts: [FinancialConnectionsPartnerAccount],
+        emailAddress: String?,
+        phoneNumber: String?,
+        country: String?,
+        consumerSessionClientSecret: String?,
+        clientSecret: String
+    ) -> Future<(
+        manifest: FinancialConnectionsSessionManifest,
+        customSuccessPaneMessage: String?
+    )> {
+        let saveAccountsToLinkHandler: () -> Future<(
+            manifest: FinancialConnectionsSessionManifest,
+            customSuccessPaneMessage: String?
+        )> = {
+            return self.saveAccountsToLink(
+                emailAddress: emailAddress,
+                phoneNumber: phoneNumber,
+                country: country,
+                selectedAccountIds: selectedAccounts.map({ $0.id }),
+                consumerSessionClientSecret: consumerSessionClientSecret,
+                clientSecret: clientSecret
+            )
+            .chained { manifest in
+                return Promise(
+                    value: (
+                        manifest: manifest,
+                        customSuccessPaneMessage: manifest.displayText?.successPane?.subCaption
+                    )
+                )
+            }
+        }
+        let linkedAccountIds = selectedAccounts.compactMap({ $0.linkedAccountId })
+        if
+            shouldPollAccounts,
+            !linkedAccountIds.isEmpty
+        {
+            return pollAccountNumbersForSelectedAccounts(
+                linkedAccountIds: linkedAccountIds,
+                clientSecret: clientSecret
+            )
+            .chained { _ in
+                return saveAccountsToLinkHandler()
+            }
+        } else {
+            return saveAccountsToLinkHandler()
+        }
+    }
+
+    private func pollAccountNumbersForSelectedAccounts(
+        linkedAccountIds: [String],
+        clientSecret: String
```

Comment:
`clientSecret` looks unused here...is it not needed for the polling?

Reply:
thank you, great catch! 

I initially added because pretty much every API call needed it - but then the API returned errors _and_ the web code showed that they exclude the client_secret too

Reply:
I removed this param in a new commit